src/lib/createDocument.ts

```diff
@@ -0,0 +1,11 @@
+import { Document, Model } from 'mongoose';
+
+export async function createDocument<T extends Document>(
```

Comment:
this method already exists inside mongoose.
model.create(anything) does exacly the same, also with the .save() call included